pkg/httpmetrics/transport.go

```diff
@@ -52,14 +52,30 @@ func SetBucketSuffixes(bs map[string]string) { bucketSuffixes = bs }
 // Transport is an http.RoundTripper that records metrics for each request.
 var Transport = WrapTransport(http.DefaultTransport)
 
+type MetricsTransport struct {
+	http.RoundTripper
```

Comment:
I don't think we need this if we have inner? Or maybe vice versa? I think it's confusing to have both

Reply:
`http.RoundTripper` is for the `outer`/`this`. 

I think there may be a better way to do this, but my Go-fu isn't as good so would love to hear how to do this better.